xmsannotator/R/compute_scores.R

```diff
@@ -1,35 +1,157 @@
+#' Compute isotopic pattern of a given molecule.
+#'
+#' @param formula A string containing molecular or empirical formula of a compound.
+#' @param minAbund A cutoff abundance ratio: only isotopes with abundance ratio above this value are returned.
+#'
+#' @return A dataframe containing isotopic pattern of a given molecule. The pattern is represented by three columns:
+#' mass, mass number difference, and abundance. Abundance of the isotopes is normalized with the most abundant isotope
+#' having a unit value and abundances of the other isotopes are represented as a share of the most abundant isotope.
+#' Mass number difference is 0 for the most abundant isotope.
+#'
+#' @export
 #' @import dplyr
-#' @importFrom rlang .data
-compute_isotopes <- function(peaks, query, rt_tolerance, mass_defect_tolerance = 0, abudance_ratio, max_isp) {
+#' @importFrom rcdk get.formula get.isotopes.pattern
+compute_isotopic_pattern <- function(formula, minAbund = 0.001) {
+  formula <- get.formula(formula)
+  isotopes <- get.isotopes.pattern(formula, minAbund)
+  isotopes <- as_tibble(isotopes)
+  isotopes <- arrange(isotopes, desc(abund))
+  isotopes <- mutate(isotopes, mass_number_difference = round(mass - mass[1]))
+}
+
+#' Match isotopes from peak table to annotated peaks based on rt cluster, rt difference, peak intensity, and mass defect.
+#'
+#' @param query A single annotated peak.
+#' @param intensity_deviation_tolerance A numeric threshold by which an intensity ratio of two isotopic peaks may differ
+#'  from their actual abundance ratio.
+#' @param peaks A peak table containing a peak identifier (unique number), mean intensity, module, and rt cluster
+#'  of each identified peak.
+#' @param mass_defect_tolerance A number. Maximum difference in mass defect between two peaks of the same compound.
+#' @param rt_tolerance A number. Maximum rt difference for two peaks of the same substance.
+#'
+#' @return A table of identified isotopes with an extra column: `mass_number_difference`.
+#'  The extra column represents a nominal mass (mass number) difference between a given isotope and the most abundant
+#'  isotope of the same molecule.
+#'
+#' @import dplyr
+filter_isotopes <- function(query,
+                            intensity_deviation_tolerance,
+                            peaks,
+                            abundance_ratio,

```

Comment:
Please add also a comment for this parameter.

Reply:
Resolved here c3bc230e362406a8856b79f522e986ef2b502d7c